trie.go

```
@@ -422,6 +435,161 @@ func (tr *Trie) matchSeq(input []byte, buf *matchBuf) {
 	}
 }
 
+// scanRange16 runs the stop-byte automaton over input[i:to), starting in
+// state s, appending matches that end at or after minEmit to raw, which
+// is returned. Positions are absolute into input; the SWAR skip may read
+// (but not emit) past to. Loads use raw pointer arithmetic; see
+// matchStopByte.
+func (tr *Trie) scanRange16(input []byte, i, to int, s uint32, minEmit int, raw []uint64) []uint64 {
+	ftBase := unsafe.Pointer(&tr.failTrans16[0])
```

golangci-lint (2.12.2)

[low] G103: Use of unsafe calls should be audited

(gosec)